app/packs/src/apps/mydb/elements/details/samples/SampleDetails.js

```diff
@@ -349,7 +349,8 @@ export default class SampleDetails extends React.Component {
   checkMolfileChange() {
     const { trackMolfile } = this.state;
     const { sample } = this.props;
-    if (trackMolfile !== sample.molfile) {
+    // !sample.isNew to allow setting mp & bp for new samples
+    if (trackMolfile !== sample.molfile && !sample.isNew) {
```

Comment:
What happens when `trackMofile` or `sample` is `undefined` or `null`?

Reply:
The trackMolfile has initial state defined. Undefined or null should not happen. Let me know if you can produce an error. If it happens, I guess if statement would not fire and nothing will happen in this case

Reply:
Ok, then it is not the case.